plugins/serializers/prometheus/convert.go

```diff
@@ -55,35 +55,30 @@ var LabelNameTable = Table{
 	},
 }
 
-func isValid(name string, table Table) bool {
-	if name == "" {
-		return false
+func isValidMetricName(name string) bool {
+	for i := 0; i < len(name); i++ {
+		b := name[i]
+		if !((b >= 'a' && b <= 'z') || (b >= 'A' && b <= 'Z') || b == '_' || b == ':' || (b >= '0' && b <= '9' && i > 0)) {
+			return false
+		}
```

Comment:
out of curiosity did you compare this to a regex comparison e.g. `[a-zA-Z_:][a-zA-Z0-9_:]*`?

Reply:
No i did not compare; i can try in a bit though

Reply:
its much more expensive actually!

```
$ go test -run=^$ -bench=. -benchtime 30s -benchmem -memprofile memprofile.out -cpuprofile profile.out
goos: linux
goarch: amd64
pkg: github.com/influxdata/telegraf/plugins/serializers/prometheusremotewrite
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkRemoteWrite-8   	   38215	    931852 ns/op	  350204 B/op	    4008 allocs/op
PASS
ok  	github.com/influxdata/telegraf/plugins/serializers/prometheusremotewrite	45.243s
```

```
(pprof) top
Showing nodes accounting for 26.77s, 49.44% of 54.15s total
Dropped 240 nodes (cum <= 0.27s)
Showing top 10 nodes out of 98
      flat  flat%   sum%        cum   cum%
     5.75s 10.62% 10.62%      5.75s 10.62%  regexp.(*bitState).shouldVisit (inline)
     5.54s 10.23% 20.85%     16.06s 29.66%  regexp.(*Regexp).tryBacktrack
     2.77s  5.12% 25.96%      2.77s  5.12%  hash/fnv.(*sum64a).Write
     2.76s  5.10% 31.06%      2.76s  5.10%  regexp/syntax.(*Inst).MatchRunePos
     2.45s  4.52% 35.59%      8.39s 15.49%  runtime.mallocgc
     1.82s  3.36% 38.95%      2.09s  3.86%  runtime.heapBitsSetType
     1.78s  3.29% 42.23%     14.14s 26.11%  github.com/influxdata/telegraf/plugins/serializers/prometheusremotewrite.(*Serializer).appendCommonLabels
     1.48s  2.73% 44.97%      1.48s  2.73%  runtime.memmove
     1.33s  2.46% 47.42%      4.14s  7.65%  runtime.scanobject
     1.09s  2.01% 49.44%      1.09s  2.01%  runtime.memclrNoHeapPointers

```

Reply:
ok thanks for trying it out!